cmd/users/main.go

```diff
@@ -292,10 +309,60 @@ func newService(db *sqlx.DB, tracer opentracing.Tracer, auth mainflux.AuthNServi
 			Help:      "Total duration of requests in microseconds.",
 		}, []string{"method"}),
 	)
-
+	if err := createAdmin(svc, userRepo, groupRepo, c); err != nil {
+		logger.Error("failed to create admin user:" + err.Error())
+		os.Exit(1)
+	}
 	return svc
 }
 
+func createAdmin(svc users.Service, userRepo users.UserRepository, groupRepo users.GroupRepository, c config) error {
+
+	user := users.User{
+		Email:    c.adminEmail,
+		Password: c.adminPassword,
+	}
+	groupName := c.adminGroup
+
+	_, err := userRepo.RetrieveByEmail(context.Background(), user.Email, false)
+	if err == nil {
```

Comment:
`RetrieveByEmail` actually fails when there is no user? I did not know that, I though it returns empty result...

Reply:
Also, I see below that you are using `users.ErrNotFound`, so please be consistent.

Reply:
Also, inline the check